package.json

```diff
@@ -41,58 +41,45 @@
   },
   "homepage": "https://github.com/recharts/react-smooth#readme",
   "peerDependencies": {
-    "react": "^15.0.0 || ^16.0.0",
-    "react-dom": "^15.0.0 || ^16.0.0"
+    "react": "^15.0.0 || ^16.0.0 || ^17.0.0",
+    "react-dom": "^15.0.0 || ^16.0.0 || ^17.0.0",
+    "prop-types": "^15.6.0"
   },
   "dependencies": {
-    "lodash": "~4.17.4",
-    "prop-types": "^15.6.0",
+    "fast-equals": "^2.0.0",
     "raf": "^3.4.0",
-    "react-transition-group": "^2.5.0"
+    "react-transition-group": "2.9.0"
   },
   "devDependencies": {
-    "@babel/cli": "^7.1.0",
-    "@babel/core": "^7.0.0",
-    "@babel/plugin-proposal-class-properties": "^7.0.0",
-    "@babel/plugin-proposal-decorators": "^7.0.0",
-    "@babel/plugin-proposal-export-default-from": "^7.0.0",
-    "@babel/plugin-proposal-export-namespace-from": "^7.0.0",
-    "@babel/plugin-proposal-function-bind": "^7.0.0",
-    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
-    "@babel/plugin-transform-runtime": "^7.0.0",
-    "@babel/polyfill": "^7.0.0",
-    "@babel/preset-env": "^7.0.0",
-    "@babel/preset-react": "^7.0.0",
-    "@babel/runtime": "^7.0.0",
+    "@babel/cli": "^7.13.0",
+    "@babel/core": "^7.13.8",
+    "@babel/plugin-proposal-class-properties": "^7.13.0",
+    "@babel/plugin-proposal-decorators": "^7.13.5",
+    "@babel/plugin-proposal-export-default-from": "^7.12.13",
+    "@babel/plugin-proposal-export-namespace-from": "^7.12.13",
+    "@babel/plugin-proposal-function-bind": "^7.12.13",
+    "@babel/plugin-proposal-object-rest-spread": "^7.13.8",
+    "@babel/plugin-transform-runtime": "^7.13.8",
+    "@babel/preset-env": "^7.13.9",
+    "@babel/preset-react": "^7.12.13",
+    "@babel/runtime": "^7.13.8",
+    "@testing-library/dom": "^7.30.0",
+    "@testing-library/jest-dom": "^5.11.9",
+    "@testing-library/react": "^11.2.5",
     "babel-eslint": "^10.0.0",
     "babel-loader": "^8.0.0",
-    "babel-plugin-lodash": "^3.3.0",
-    "chai": "^4.1.2",
-    "chai-enzyme": "^0.8.0",
-    "chai-spies": "^0.7.1",
+    "core-js": "^3.9.1",
```

Comment:
Unused?

Reply:
What's unused? Corejs is required by presetenv

Reply:
I gather it has to be explicitly enabled, and is disabled by default:

https://babeljs.io/docs/en/babel-preset-env#usebuiltins

(Which is good. Polyfill = application responsibility, not library)

Reply:
That's not 100% right, on recharts we've been shipping library compiled with env to be compatible with at least ie11 and last 2 versions.

Reply:
I noticed, and I think it's a bit unusual. The use of `Webpack` itself here is a bit unusual. It's a tool for bundling client js for applications. The underlying `babel` can/should be used by itself when we are creating libraries. Under the covers, webpack is just calling upon `babel` to do the transpile.

[axios](https://github.com/axios/axios) is another famous project for promulgating this approach, which has sharp edges.

How do large projects end up committing seemingly obvious errors?

Consider the timeline: in 2016 and earlier, we had very limited choices for putting together demos/docs. Naturally, `webpack` can do this for us, but some project(s) end up using webpack for the _build_ too, which is really better left for `babel` alone to do.

----

Seen from an application developers' perspective: "Oh, guess I picked up another `Promise` implementation." or "Here goes another way to do `Object.assign`."  The developer now needs to a) address bloat if there is a lot of polyfill cruft, b) worry about conflicts of interest in competing polyfill solutions.

The _application developer_ needs to be in charge of their target environment(s). If _they_ need a polyfill, they should add it...  we cannot possibly imagine all the possible environments the code could be running in. And it's simple enough for the consumer to just add what they need.

Reply:
To get back on topic: @semoal

For this PR: I think it's fine if you need to add `core-js` (it was in recharts main repo at some point), but prefer to leave it out if possible.

I will write a more detailed post about this problem soon, as I haven't seen this problem described in any detail on the web. Sorry for the rant.